### PR TITLE
Update dependencies to work with Rails 5

### PR DESCRIPTION
**Pull request by schuylr**
Addresses #31 - seems like it works as desired right out of the box. Here's how I tested compatibility with Rails 5.0.0.beta1.1:

```
gem install rails --pre
rails new rails5
cd rails5
bin/rails g controller Main
```

Add `get '/', :to => 'main#show'` to `routes.rb`

MainController.rb:

```
class MainController < ApplicationController
  caches_page :show

  def show
  end
end
```

app/views/main/show.html.erb:

```
Hello, world.
```

Then `RAILS_ENV=production bin/rails s`, go to `localhost:3000`. Then I found this in `public/index.html`:

```
<!DOCTYPE html>
<html>
  <head>
    <title>Rails5</title>
    <meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="..." />
    <meta name="action-cable-url" content="/cable" />

    <link rel="stylesheet" media="all" href="/stylesheets/application.css" data-turbolinks-track="true" />
    <script src="/javascripts/application.js" data-turbolinks-track="true"></script>
  </head>

  <body>
    Hello, world.

  </body>
</html>
```

and on `production.log`:

```
I, [2016-01-25T16:35:45.127717 #33086]  INFO -- : Started GET "/" for 127.0.0.1 at 2016-01-25 16:35:45 -0500
I, [2016-01-25T16:35:45.131745 #33086]  INFO -- : Processing by MainController#show as HTML
I, [2016-01-25T16:35:45.142266 #33086]  INFO -- :   Rendered main/show.html.erb within layouts/application (0.3ms)
I, [2016-01-25T16:35:45.147295 #33086]  INFO -- : Write page /Users/sjagr/GitRepos/rails5/public/index.html (0.3ms)
I, [2016-01-25T16:35:45.147512 #33086]  INFO -- : Completed 200 OK in 16ms (Views: 6.9ms | ActiveRecord: 0.0ms)
```

I took the liberty of bumping the version to 1.0.3 and adding a note to the CHANGELOG as well. I'm wondering if any new tests would be necessary with newer Ruby and Rails versions.

### CODE REVIEW

**Comment by schuylr** [comment]
I'm seeing lots of deprecation warnings in the tests so I'll start looking at those

**Comment by schuylr** [comment]
Issue seems to be from [this commit](https://github.com/rails/rails/commit/3806eb70a63f0252c88ebb25f0aa79b4c0ca5686) - using `create` instead should work for `ActionController::TestRequest`

**Comment by schuylr** [comment]
Note this fails only for Gemfile-4-0-stable due to a lack of `:html` and `:body` options for the `render` method. Details outlined in #31

File: test/log_subscriber_test.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class LogSubscribersController < ActionController::Base
 
     def with_page_cache
       cache_page('Super soaker', '/index.html')
-      render nothing: true
+      render body: nil
```
Comment (eileencodes) [inline test/log_subscriber_test.rb]: This should be `head :ok` per the deprecation warning: DEPRECATION WARNING: `:nothing` option is deprecated and will be removed in Rails 5.1. Use `head` method to respond with empty response body

**Comment by eileencodes** [comment]
For a lot of the other peripheral gems we've been dropping Rails versions that don't work and splitting the gems into a 1.0 and 2.0 (rails 5+). If there's no easy way to get Rails 4.0 working bump the version to 2.0.0 and remove the Rails 4 gemfiles from travis. 

We can then make a 1-0-stable branch off master before this is merged for changes needed to pre-Rails 5.

File: test/log_subscriber_test.rb (hunk 1 above)
Comment (schuylr) [inline test/log_subscriber_test.rb]: @eileencodes Sure thing! The deprecation message mentioned either `body: nil` or `head :ok` would work in lieu of `nothing: true` - but I will opt for the latter. Either one broke Rails 4.0 anyways.

**Comment by schuylr** [comment]
@eileencodes Gladly! I'll see if I can get this done today.

**Comment by enrico** [comment]
are there any plans to merge this in now that rails 5 is officially out?

**Comment by rafaelfranca** [comment]
Closing in favor of https://github.com/rails/actionpack-page_caching/pull/41

**Pull request closed without merging**